geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/uuid/Z3FeatureIdGenerator.scala

```diff
@@ -75,7 +77,13 @@ object Z3UuidGenerator extends RandomLsbUuidGenerator {
     createUuid(geom.safeCentroid(), time, period)
   }
 
-  def createUuid(pt: Point, time: Long, period: TimePeriod) = {
+  def createUuid(pt: Point, time: Long, period: TimePeriod): UUID = {
+    // handle missing geometry
+    if (pt == null) {
+      logger.error("Cannot generate a meaningful UUID for a <NULL> geometry; returning a random UUID instead.")
```

Comment:
I'm of half a mind.  On the one hand, a warning is probably appropriate (as we're returning a valid UUID for this feature, even though it has a null geometry).  On the other hand, that we are enabling users to create a feature with an invalid geometry feels... wrong.  Should this entire PR be retracted?  Was the original behavior correct?  Or should we refactor it to still throw an exception, just one with a nicer error?  (Either way, I like having the additional unit test so that the expectation is clear.)